Documentation/api.md

```diff
@@ -1,5 +1,27 @@
 # etcd API
 
+- [Key Space Operations](#key-space-operations)
```

Comment:
I also think it contains too many items and hard to follow.
A good categorized API so far is here: https://github.com/coreos/etcd/blob/master/client/keys.go#L93
It lists the most common operations at the first level, and give more options in the secondary level.